onnxruntime/core/optimizer/gemm_activation_fusion.cc

```diff
@@ -66,8 +66,9 @@ Status GemmActivationFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
 
     //Add optional attributes for activations
     if (act_node.OpType() == "LeakyRelu") {
-      const NodeAttributes& attrs = act_node.GetAttributes();
-      for (const auto& attr : attrs) {
+      NodeAttributes& attrs = const_cast<NodeAttributes&>(act_node.GetAttributes());
+      for (auto& attr : attrs) {
```

Comment:
Please avoid const_cast wherever possible. It destroys const correctness. If the Node API didn't intend for you to modify the attributes (the Node::Init takes const Attributes, which implies there was never an intention for Node to be able to modify them or own them), using const_cast to work around that means you are changing code in potentially unexpected ways, which will potentially break in unexpected places.

Given you're calling AddAttribute with the updated value, can you instead copy-construct a new attribute from the const instance of the existing one?

Not sure you can given the Node AddAttribute API takes a const reference, so the ownership of the Attribute is unclear. Which also makes this code suspect as you're modifying an attribute that came from act_node, putting it in fused_gemm, but neither of those appear to own the attribute.